logging/logger.go

```diff
@@ -0,0 +1,77 @@
+package logging
+
+import (
+	"context"
+)
+
+var (
+	// DefaultLogger is the default Logger for all SDK logging, if one hasn't been provided in the context.
+	DefaultLogger Logger = &NoOpLogger{}
+
+	contextKey = loggerContextKey{}
+)
+
+type loggerContextKey struct{}
+
+// FromContext returns the Logger set in the context with Context(), or the DefaultLogger if no Logger is set in the context.
+// If DefaultLogger is nil, it returns a *NoOpLogger so that the return is always valid to call methods on without nil-checking.
+// So long as the Logger to return is not the NoOpLogger, it will call WithContext(ctx) on the logger in the context and return
+// the resulting Logger.
+func FromContext(ctx context.Context) Logger {
+	l := ctx.Value(contextKey)
+	if l != nil {
+		if logger, ok := l.(Logger); ok {
+			return logger.WithContext(ctx)
+		}
+	}
+	if DefaultLogger != nil {
+		return DefaultLogger.WithContext(ctx)
+	}
+	return &NoOpLogger{}
+}
+
+// Context returns a new context built from the provided context with the provided logger in it.
+// The Logger added with Context() can be retrieved with FromContext()
+func Context(ctx context.Context, logger Logger) context.Context {
+	return context.WithValue(ctx, contextKey, logger)
+}
+
+// Logger extends the BasicLogger interface with methods that accept a context as well as the message and key/val args.
```

Comment:
Should this be rephrased as:
- I can't see `BasicLogger`
- Methods carrying context are not there anymore

Reply:
Good call, I'll update the godocs!